### PR TITLE
fix: replace_time_zone with single-null-element "ambiguous" was panicking

### LINKED ISSUE #14970: replace_time_zone with ambiguous with single null value panics
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl
from datetime import datetime
df = pl.DataFrame(
    {
        "ts": [datetime(2020, 10, 2, 1, 1)],
        "ambiguous": [None],
    },
    schema_overrides={'ambiguous': pl.String}
)

result = df.select(
    pl.col("ts").dt.replace_time_zone(
        "Europe/London", ambiguous=pl.col("ambiguous")
    )
)["ts"]
print(result)
```

### Log output

```shell
thread '<unnamed>' panicked at crates/polars-ops/src/chunked_array/datetime/replace_time_zone.rs:71:76:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Traceback (most recent call last):
  File "/home/marcogorelli/scratch/t.py", line 11, in <module>
    result = df.select(
             ^^^^^^^^^^
  File "/home/marcogorelli/scratch/.venv/lib/python3.11/site-packages/polars/dataframe/frame.py", line 8116, in select
    return self.lazy().select(*exprs, **named_exprs).collect(_eager=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marcogorelli/scratch/.venv/lib/python3.11/site-packages/polars/lazyframe/frame.py", line 1934, in collect
    return wrap_df(ldf.collect())
                   ^^^^^^^^^^^^^
pyo3_runtime.PanicException: called `Option::unwrap()` on a `None` value
```

### Issue description

it panics

### Expected behavior

it should just return a null

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.14
Index type:           UInt32
Platform:             Linux-5.15.146.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Python:               3.11.7 (main, Dec  8 2023, 18:56:58) [GCC 11.4.0]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          <not installed>
connectorx:           <not installed>
deltalake:            <not installed>
fastexcel:            <not installed>
fsspec:               <not installed>
gevent:               <not installed>
hvplot:               0.9.2
matplotlib:           <not installed>
numpy:                1.26.4
openpyxl:             <not installed>
pandas:               2.2.1
pyarrow:              15.0.0
pydantic:             2.6.1
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           <not installed>
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>```

</details>

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #14970

### CODE REVIEW

**Review by stinodego — COMMENTED** [review]
This case should throw an error, right? `null` is not a valid input for `ambiguous`.

File: crates/polars-ops/src/chunked_array/datetime/replace_time_zone.rs [hunk 1]
```diff
@@ -68,7 +68,10 @@ pub fn replace_time_zone(
         }),
     };
     let mut out = out?.into_datetime(datetime.time_unit(), time_zone.map(|x| x.to_string()));
-    if from_time_zone == "UTC" && ambiguous.len() == 1 && ambiguous.get(0).unwrap() == "raise" {
+    if from_time_zone == "UTC"
+        && ambiguous.len() == 1
+        && unsafe { ambiguous.get_unchecked(0) } == Some("raise")
```
Comment (ritchie46) [inline crates/polars-ops/src/chunked_array/datetime/replace_time_zone.rs:73]: I don't think we should use `unsafe` here. This isn't called repeatedly in a hot loop.

**Comment by MarcoGorelli** [comment]
> This case should throw an error, right? `null` is not a valid input for `ambiguous`.

`ambiguous` can take an expression, e.g.:
```python
In [4]: df
Out[4]:
shape: (4, 2)
┌─────────────────────┬───────────┐
│ ts                  ┆ ambiguous │
│ ---                 ┆ ---       │
│ datetime[μs]        ┆ str       │
╞═════════════════════╪═══════════╡
│ 2018-10-28 01:30:00 ┆ earliest  │
│ 2018-10-28 02:00:00 ┆ earliest  │
│ 2018-10-28 02:30:00 ┆ latest    │
│ 2018-10-28 02:00:00 ┆ latest    │
└─────────────────────┴───────────┘

In [5]: df.with_columns(
   ...:     ts_localized=pl.col("ts").dt.replace_time_zone(
   ...:         "Europe/Brussels", ambiguous=pl.col("ambiguous")
   ...:     )
   ...: )
Out[5]:
shape: (4, 3)
┌─────────────────────┬───────────┬───────────────────────────────┐
│ ts                  ┆ ambiguous ┆ ts_localized                  │
│ ---                 ┆ ---       ┆ ---                           │
│ datetime[μs]        ┆ str       ┆ datetime[μs, Europe/Brussels] │
╞═════════════════════╪═══════════╪═══════════════════════════════╡
│ 2018-10-28 01:30:00 ┆ earliest  ┆ 2018-10-28 01:30:00 CEST      │
│ 2018-10-28 02:00:00 ┆ earliest  ┆ 2018-10-28 02:00:00 CEST      │
│ 2018-10-28 02:30:00 ┆ latest    ┆ 2018-10-28 02:30:00 CET       │
│ 2018-10-28 02:00:00 ┆ latest    ┆ 2018-10-28 02:00:00 CET       │
└─────────────────────┴───────────┴───────────────────────────────┘
```
If one of those elements is missing, then the null value propagates:
```python
In [7]: df
Out[7]:
shape: (4, 2)
┌─────────────────────┬───────────┐
│ ts                  ┆ ambiguous │
│ ---                 ┆ ---       │
│ datetime[μs]        ┆ str       │
╞═════════════════════╪═══════════╡
│ 2018-10-28 01:30:00 ┆ earliest  │
│ 2018-10-28 02:00:00 ┆ earliest  │
│ 2018-10-28 02:30:00 ┆ latest    │
│ 2018-10-28 02:00:00 ┆ null      │
└─────────────────────┴───────────┘

In [8]: df.with_columns(
   ...:     ts_localized=pl.col("ts").dt.replace_time_zone(
   ...:         "Europe/Brussels", ambiguous=pl.col("ambiguous")
   ...:     )
   ...: )
Out[8]:
shape: (4, 3)
┌─────────────────────┬───────────┬───────────────────────────────┐
│ ts                  ┆ ambiguous ┆ ts_localized                  │
│ ---                 ┆ ---       ┆ ---                           │
│ datetime[μs]        ┆ str       ┆ datetime[μs, Europe/Brussels] │
╞═════════════════════╪═══════════╪═══════════════════════════════╡
│ 2018-10-28 01:30:00 ┆ earliest  ┆ 2018-10-28 01:30:00 CEST      │
│ 2018-10-28 02:00:00 ┆ earliest  ┆ 2018-10-28 02:00:00 CEST      │
│ 2018-10-28 02:30:00 ┆ latest    ┆ 2018-10-28 02:30:00 CET       │
│ 2018-10-28 02:00:00 ┆ null      ┆ null                          │
└─────────────────────┴───────────┴───────────────────────────────┘
```
So if there's a single null value, then that should also propagate. However, on the latest release, it panics:
```
In [9]: df[3:]
Out[9]:
shape: (1, 2)
┌─────────────────────┬───────────┐
│ ts                  ┆ ambiguous │
│ ---                 ┆ ---       │
│ datetime[μs]        ┆ str       │
╞═════════════════════╪═══════════╡
│ 2018-10-28 02:00:00 ┆ null      │
└─────────────────────┴───────────┘

In [10]: df[3:].with_columns(
    ...:     ts_localized=pl.col("ts").dt.replace_time_zone(
    ...:         "Europe/Brussels", ambiguous=pl.col("ambiguous")
    ...:     )
    ...: )
thread '<unnamed>' panicked at crates/polars-ops/src/chunked_array/datetime/replace_time_zone.rs:71:76:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
---------------------------------------------------------------------------
[...]

PanicException: called `Option::unwrap()` on a `None` value
```

File: crates/polars-ops/src/chunked_array/datetime/replace_time_zone.rs (hunk 1 above)
Comment (MarcoGorelli) [reply crates/polars-ops/src/chunked_array/datetime/replace_time_zone.rs]: right, thanks

**Comment by stinodego** [comment]
Thanks for the explanation - I didn't know a null would propagate here.

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**